### DISCUSSION TITLE
v2.0.1: Entity update error when _0d expires

### DISCUSSION THREAD

**Discussion by genem2** [cytech/Home-Assistant-wundergroundpws#172] [Q&A] [answered]
I could use some direction on this:
Home-Assistant: 2023.5.3 (container)
Wundergroundpws: v2.0.1

I reverted my HA back to 2022.4.6 and verified that my WU Sensors matched the documentation:
_> API after 3:00pm Local Apparent Time. The affected sensors will return as "Today Expired" with a value of "—" when this condition is met._
```
entity: sensor.kcasanjo1654_forecast_temperature_0d
Friendly Name: Forecast Temperature Today Expired
State: —
unit_of_measurement: 
attribution: Data provided by the WUnderground weather service
device_class: temperature
icon: mdi:thermometer
friendly_name: Forecast Temperature Today Expired
```

But on 2023.5.3, all is fine until 15:00, when my log is bombarded with:

```
2023-05-15 16:02:49.128 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform wundergroundpws
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 579, in state
    numerical_value = int(value)
ValueError: invalid literal for int() with base 10: '—'
```

At this time, and until _0d returns the following day, the entity _state_ (developer-tools/state or from a template render) remains as the previous numeric value, and friendly_name remains as Forecast Temperature Today. (Makes sense I guess, given the entity update failed.)  If I restart HA, it reports as state:unavailable friendly_name:Forecast Temperature Today Expired.

So, would this be a custom_component issue, or something for HA?
Many thanks...

**Comment by genem2** [comment]
Well, it appears HA doesn't want Strings returned for these type of sensors:
`ValueError: Sensor sensor.kcasanjo1654_forecast_temperature_0d has device class temperature, state class None unit  and suggested precision None thus indicating it has a numeric value; however, it has the non-numeric value: — (<class 'str'>)
`

I don't know the proper way to resolve this, but as a temporary workaround:
I replaced all instances of `else str('—'),`
with: `else int(0),` 
in my local `custom_components\wundergroundpws\wupws_forecast_sensors.py` file.

_I guess this should be moved out of Discussions to Issues?_

**Comment by cytech** [comment, marked as answer]
its already a known issue
at this point it is just a warning and goes away after a cycle until the next time HA is restarted

**Reply by genem2** [reply]
Apologies - can't believe I missed it...
FWIW: I'm not seeing it go away; it's 25 lines in the log every 5 min from 15:00-03:00.
Anyway, my workaround holds me until #169  is resolved.  Tnx

**Comment by clau-bucur** [comment]
@cytech I've opened a bug ticket on this: https://github.com/cytech/Home-Assistant-wundergroundpws/issues/173
For me at least seems not to be a warning and it's persistent between restarts to the point that the _0d sensors got disabled and are now unavailable:
![image](https://github.com/cytech/Home-Assistant-wundergroundpws/assets/47672949/5440d41c-1c9e-4015-ae2b-9715dc33bb7b)

**Comment by cytech** [comment]
issue fixed in v2.0.2
